stdpopsim/catalog/BosTau/demographic_models.py

```diff
@@ -1,31 +1,34 @@
 import msprime
-
 import stdpopsim
 
 _species = stdpopsim.get_species("BosTau")
 
 # Inferring Demography from Runs of Homozygosity in Whole-Genome Sequence,
 # with Correction for Sequence Errors.
 _MacLeodEtAl = stdpopsim.Citation(
-    doi="https://doi.org/10.1093/molbev/mst125",
-    year="2013",
     author="MacLeod et al.",
+    year=2013,
+    doi="https://doi.org/10.1093/molbev/mst125",
+    reasons={
+        stdpopsim.CiteReason.GEN_TIME,
+        stdpopsim.CiteReason.POP_SIZE,
+        stdpopsim.CiteReason.DEM_MODEL,
+    },
```

Comment:
I think for this citation, it's better to have the reasons given in the different locations where it's used, with the `.because(...)` form. It is possible someone will use the species definition with a different demographic model (or a constant size model), in which case the citation reasons shouldn't include DEM_MODEL.

I.e. The "please cite" output for running:
```
python -m stdpopsim BosTau -D -c 1 10
```
Shouldn't mention "demographic model" for the McLeod citation.

Reply:
Done!